protocol/pubsub/v2/message.go

```diff
@@ -120,14 +121,14 @@ func (m *Message) GetExtension(name string) interface{} {
 }
 
 // Finish marks the message to be forgotten.
-// If err is nil, the underlying Pubsub message will be acked;
-// otherwise nacked and return the error.
+// Regarding protocol.IsAck, the Pubsub message will be acked or not.
+// If not acked, the original error will be returned.
 func (m *Message) Finish(err error) error {
-	if err != nil {
-		m.internal.Nack()
-		return err
+	if protocol.IsACK(err) {
+		m.internal.Ack()
+		return nil
```

Comment:
question: this would mean we're dropping any potential error if `err != nil` here. Would there be a risk of losing error information then? Why not `return err` here like you do in the other case below?

Reply:
Hi @embano1,
When an error is returned here, it's forwarded as the result of the "invoke" func of the client. In that case, it just triggers some logs: https://github.com/cloudevents/sdk-go/blob/main/v2/client/client.go#L261
The error returned remains available when wrapping receiver using RecordCallingInvoker https://github.com/cloudevents/sdk-go/blob/main/v2/client/invoker.go#L91 (the defer Finish is not called at this point).
So I'm assuming the developer made an assumed choice by using a protocol.Result rather than an error and he probably don't want to be anoyed by some warning logs he may probably handle himself in the receive func or in custom callinginvoker hook (the choice made here is not 100% sure to me but I think it may the better one in that case).

Reply:
IMHO the current change is a breaking change and we shouldn't do it (I don't see a strong reason to change error handling here). So let's please `return err` here as well (`err` could be `nil` of course, which also handles `nil` scenarios). Please add a test to assert this.

Reply:
Done ✅